src/super_gradients/training/utils/checkpoint_utils.py

```diff
@@ -156,6 +161,1229 @@ def read_ckpt_state_dict(ckpt_path: str, device="cpu") -> Mapping[str, torch.Ten
         return state_dict
 
 
+class DefaultCheckpointSolver:
+    """
+    Default checkpoint solver class - has same behavior as PyTorch's default checkpoint solver.
+    """
+
+    def __call__(self, model_state_dict: Mapping[str, Tensor], checkpoint_state_dict: Mapping[str, Tensor]) -> Mapping[str, Tensor]:
+        new_ckpt_dict = {}
+        for (ckpt_key, ckpt_val), (model_key, model_val) in zip(checkpoint_state_dict.items(), model_state_dict.items()):
+
+            if ckpt_val.shape != model_val.shape:
+                raise ValueError(f"ckpt layer {ckpt_key} with shape {ckpt_val.shape} does not match {model_key}" f" with shape {model_val.shape} in the model")
+            new_ckpt_dict[model_key] = ckpt_val
+        return {"net": new_ckpt_dict}
+
+
+class YoloXCheckpointSolver:
+    @classmethod
+    def generate_mapping_table(cls) -> Mapping[str, str]:
+        """
+        Helper method to generate mapping table between olx YoloX checkpoints and the current YoloX layer names.
+        :return: A mapping dictionary {checkpoint_key: model_key}
+        """
+        from super_gradients.common.object_names import Models
+        from super_gradients.training import models
```

Comment:
Is there a motivation to not move these imports to the top of the module?

Reply:
Yes, this is to avoid circular imports.